core/src/test/java/org/elasticsearch/search/builder/SearchSourceBuilderTest.java

```diff
@@ -0,0 +1,89 @@
+import org.elasticsearch.common.io.stream.NamedWriteableRegistry;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.indices.IndicesModule;
+import org.elasticsearch.search.SearchModule;
+import org.elasticsearch.search.suggest.SuggestBuilder;
+import org.elasticsearch.search.suggest.phrase.PhraseSuggestionBuilder;
+import org.elasticsearch.test.ESTestCase;
+import org.junit.BeforeClass;
+
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Collections.emptyList;
+import static org.elasticsearch.search.builder.SearchSourceBuilderTests.createSearchSourceBuilder;
+import static org.elasticsearch.search.suggest.AbstractSuggestionBuilderTestCase.setCommonPropertiesOnRandomBuilder;
+
+public class SearchSourceBuilderTest extends ESTestCase {
+    private static NamedWriteableRegistry namedWriteableRegistry;
+
+    @BeforeClass
+    public static void beforeClass() {
+        IndicesModule indicesModule = new IndicesModule(emptyList()) {
+            @Override
+            protected void configure() {
+                bindMapperExtension();
+            }
+        };
+        SearchModule searchModule = new SearchModule(Settings.EMPTY, false, emptyList()) {
+            @Override
+            protected void configureSearch() {
+                // Skip me
+            }
+        };
+        List<NamedWriteableRegistry.Entry> entries = new ArrayList<>();
+        entries.addAll(indicesModule.getNamedWriteables());
+        entries.addAll(searchModule.getNamedWriteables());
+        namedWriteableRegistry = new NamedWriteableRegistry(entries);
+    }
+
+    public void testSearchRequestBuilderSerialization() throws Exception {
+        SearchSourceBuilder searchSourceBuilder = createSearchSourceBuilderWithPhraseSuggestionBuilder();
+        try (BytesStreamOutput output = new BytesStreamOutput()) {
+            searchSourceBuilder.writeTo(output);
+            try (StreamInput in = new NamedWriteableAwareStreamInput(output.bytes().streamInput(), namedWriteableRegistry)) {
+                SearchSourceBuilder deserializedSearchSourceBuilder = new SearchSourceBuilder(in);
+                BytesStreamOutput deserializedOutput = new BytesStreamOutput();
+                deserializedSearchSourceBuilder.writeTo(deserializedOutput);
+                Iterator<Map.Entry<String, Object>> streamInMap =
+                    ((PhraseSuggestionBuilder) searchSourceBuilder.suggest().getSuggestions().get("suggestion1")).collateParams().entrySet().iterator();
+                Iterator<Map.Entry<String, Object>> streamOutMap =
+                    ((PhraseSuggestionBuilder) deserializedSearchSourceBuilder.suggest().getSuggestions().get("suggestion1")).collateParams().entrySet().iterator();
+                assertMapsOrder(streamInMap, streamOutMap);
```

Comment:
Thanks for this - it'll be a lot easier to debug failures than to look at two differing byte arrays and back into the difference.